core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -60,11 +61,12 @@ class ClientContext(masters: Iterable[HostPort]) {
 
   def submit(app : Application, jarPath: String) : Int = {
     val client = new MasterClient(master)
+    val updatedApp = resetAppName(app, System.getProperty(GEARPUMP_APP_NAME_PREFIX))
```

Comment:
How about this: 

val newName = checkAndAddNamePrefix(inputApp.name)
val app = appInput.copy(name = newName)